glfw/src/main.zig

```diff
@@ -33,6 +33,7 @@ pub usingnamespace @import("vulkan.zig");
 pub usingnamespace @import("time.zig");
 pub usingnamespace @import("hat.zig");
 pub usingnamespace @import("mod.zig");
+pub const native = @import("native.zig");
```

Comment:
Here's an idea, not 100% positive it'll work but I think so and may be a much nicer way to do this?

We could expose different imports here like `native_win32`, `native_wgl`, `native_cocoa`, `native_nsgl`, etc.

Inside of each of those, we could instead of `const c = @import("c.zig").c;` directly use `@cInclude("GLFW/glfw3native.h");` and `@cDefine` the right definitions.

Then only if someone references that code (`glfw.native_win32`, etc.) within theirs would Zig actually interpret the `@cInclude` in that `native_win32.zig` file, I believe, and so no special configuration or platform detection is needed (it's left up to the user)

Reply:
I actually tried something similar. The problem is that a @cImport containing @cInclude of glfw3native.h also needs to include glfw3.h, and doing that would create incompatibilities with c.zig's cImport.

One way to go around it is by @ptrCast-ing them, and that this guaranteed to work (despite looking ugly) but I would need some confirmation first.

To explain more, the way it is designed is, you use the API as follow:
```zig
const native = glfw.Native(.{ .x11 = true, .glx = true });
const xwindow = native.getX11Window(window); 
```

On the impl side:
```zig
pub fn Native(comptime options: Options) type {
    const native = @cImport({
        // include glfw3.h first because its needed
        if (options.x11) // define x11 flag
        // include glfw3native.h
    });
    return struct {
        pub fn getX11Window(window: Window) u32 {
            return @intCast(u32, native.glfwGetX11Window(@ptrCast(*native.GLFWwindow, window.handle)));
        }
    };
}
```

The reason why I am suggesting arbitary backend combination because certain backends like egl and osmesa works on all platform, so its up the the user to decide.

Reply:
Got it. I think I am OK with both your solution above, as well as @ptrCast'ing approach so whichever you are happier with sounds great to me!